src/components/selection-list/selection-list-model.tsx

```diff
@@ -0,0 +1,159 @@
+import {extendShallowObservable} from 'mobx';
+import React = require('react');
+import {clamp} from '../../utils/clamp';
+import {divider, Divider} from './divider';
+import {Option} from './option';
+
+export type DataSourceItem = string | object | symbol;
+
+export interface OptionList {
+    dataSource?: DataSourceItem[];
+    dataSchema?: {};
+    renderItem?: (item: DataSourceItem) => React.ReactElement<any> | null;
+}
+
+export type SelectionListItemValue = string;
+export type SelectionListItemData = DataSourceItem | React.ReactElement<any>;
+
+export interface SelectionListItem {
+    data: SelectionListItemData;
+    disabled: boolean;
+    element: React.ReactElement<any>;
```

Comment:
In places where you dont need the `| null` i think `JSX.Element` would be more readable, and the same, as it is defined
```
namespace JSX {
        interface Element extends React.ReactElement<any> { }
```